### PR TITLE
API/REF: Standardize examples.

### PR DESCRIPTION
**Pull request by danielballan**
- Move examples that are in doc/source/examples into package.
- *Shorten* some names. (Yes, you read that right.)
- Make a prefix command-line argument optional.
- Consolidate `main()` into `__main__` to simplify things.

### CODE REVIEW

File: caproto/curio/server.py [hunk 1]
```diff
@@ -32,7 +32,6 @@ class AsyncLibraryLayer:
 
 class CurioAsyncLayer(AsyncLibraryLayer):
     name = 'curio'
-    sleep = curio.sleep
```
Comment (tacaswell) [inline caproto/curio/server.py]: no naps?

Comment (tacaswell) [reply caproto/curio/server.py]: ah, I see, expect the library to provide `sleep`

Comment (klauer) [reply caproto/curio/server.py]: Fix here could be `sleep = staticmethod(curio.sleep)` if we think it's desirable to increase the scope of this 'async library layer'

Comment (tacaswell) [reply caproto/curio/server.py]: no, I think we should only shim out when we have to and I strongly suspect everyone has 'sleep'.

**Pull request merged**